src/Microsoft.Azure.WebJobs.Host/TypeUtility.cs

```diff
@@ -74,5 +75,29 @@ internal static bool IsJObject(Type type)
 
             return null;
         }
+
+        public static bool IsAsync(MethodInfo methodInfo)
+        {
+            if (methodInfo == null)
+            {
+                throw new ArgumentNullException(nameof(methodInfo));
+            }
+
+            var stateMachineAttribute = methodInfo.GetCustomAttribute<AsyncStateMachineAttribute>();
```

Comment:
Standard attributes applied by the compiler. E.g. http://stackoverflow.com/questions/13183313/how-to-determine-if-method-is-async-at-runtime